src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverProjectExtension.kt

```diff
@@ -70,6 +70,12 @@ public interface KoverProjectExtension {
      */
     public fun excludeInstrumentation(config: Action<KoverInstrumentationExclusions>)
 
+    /**
+     * Configure default Kover artifact.
+     *
+     * Affects the default reports of this project and any project that specifies this project as a dependency.
+     */
+    public fun default(config: Action<DefaultArtifactConfigs>)
```

Comment:
I see the following issues with the API shape:

1) It introduces a notion of `Kover artifact`, but it's not a JAR-like artifact, but rather an output of the Kover task. Maybe it should be named such?

2) `default` has an arbitrary meaning and it's really hard to grasp what it does without diving into Android-specific examples, which is a bummer.
Please write extensive documentation (ideally with built-in examples similar to how we documented verification rules); maybe a more natural name will appear (e.g. `defaultKoverDependencies` or `andoid {}`)

Reply:
1. it's Gradle-based notion (see [Gradle Component Model](https://docs.gradle.org/current/userguide/variant_model.html#gradle_component_model) and [Configurations](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.Configuration.html)). Artifacts - is a files or directories shared between projects by dependencies. JAR-like artifact - is artifact produced by `jar` Gradle task.

Reply:
2. It might be worth reworking the DSL a bit - it won't take much time. However, I agree, in this case, it is not quite obvious how to do this or that action.

Reply:
WDYT about this DSL? 

```
kover {
	// disable instrumentation of current projects, reports task don't trigger it's test tasks, reports not generated if no deps
	disable()

	// use specified toolset for collecting coverage and generating reports
	use...Tool()

	// specify the coverage of which data should be measured and by what
	measuring {
		// java from this project will be skipped 
		excludeJavaCode = true

		// exclude specified classes from instrumentation: coverage for this classes will be 0
		excludeInstrumentation {
			...
		}

		// exclude specified Kotlin compilations (source sets for K/JVM, compilations for K/MPP)
		excludeCompilation {
			...
		}

		// ignore code calls that occur when performing the specified test tasks
		excludeTest {
			...
		}
	}
}



koverReport {
	// common filters for reports of all variants
        filters {}

	// settings of reports for default variant (K/JVM or K/MPP code)
	defaultReports {
		// add data from the specified named report to the default reports
		mergeWith(otherVariant)

		filters {}
		xml {}
		html {}
		verify {}
	}

	// setting of reports only for specified android build variant
	// or androidReports(variant) ??
	reports(variant) {
		filters {}
		xml {}
		html {}
		verify {}
	}
}
```